src/unsorted_list/mod.rs

```diff
@@ -0,0 +1,254 @@
+use std::default::Default;
+use std::iter::FromIterator;
+use std::ops::{Index, IndexMut};
+
+const DEFAULT_LOAD_FACTOR: usize = 1000;
+
+#[derive(Debug)]
+pub struct UnsortedList<T> {
+    lists: Vec<Vec<T>>, // There is always at least one element in the outer list.
+    load_factor: usize,
+    len: usize,
+}
+
+impl<T> UnsortedList<T> {
+    pub fn insert(&mut self, mut i: usize, element: T) {
+        let mut outer = 0;
+        // biases towards the earlier list.
+        while i > self.lists[outer].len() {
+            i -= self.lists[outer].len();
+            outer += 1;
+        }
+
+        self.lists[outer].insert(i, element);
+        self.len += 1;
+        self.expand(outer);
+    }
+
+    /// Splits sublists that are more than double the load level.
+    /// Updates the i when the sublist length is less than double the load
```

Comment:
This should be "updates the index". And I think the parameter should be named `index`.